src/database/MySQLTable.ts

```diff
@@ -83,17 +83,20 @@ export class MySQLTable {
     }
   }
 
-  private query(type: RecordAction, rows: ParsedRecord[]): Promise<void> {
+  private async query(type: RecordAction, rows: ParsedRecord[]): Promise<void> {
     const rowValues = rows.map(r => Object.values(r.values));
 
     switch (type) {
       case RecordAction.Insert:
       case RecordAction.DelayedInsert:
-        return this.db.query(`INSERT IGNORE INTO \`${this.tableName}\` VALUES ?`, [rowValues]);
+        await this.db.query(`INSERT IGNORE INTO \`${this.tableName}\` VALUES ?`, [rowValues]);
+        return;
       case RecordAction.Update:
-        return this.db.query(`REPLACE INTO \`${this.tableName}\` VALUES ?`, [rowValues]);
+        await this.db.query(`REPLACE INTO \`${this.tableName}\` VALUES ?`, [rowValues]);
+        return;
       case RecordAction.Delete:
-        return this.db.query(`DELETE FROM \`${this.tableName}\` WHERE (${this.getDeleteSQL(rows)})`, rows.flatMap(row => Object.values(row.keysValues)));
+        await this.db.query(`DELETE FROM \`${this.tableName}\` WHERE (${this.getDeleteSQL(rows)})`, rows.flatMap(row => Object.values(row.keysValues)));
+        return;
```

Comment:
same as above, this doesn't do anything different and I'd argue it's a bit more verbose